zerver/lib/request.py

```diff
@@ -24,6 +27,43 @@
 
 from zerver.lib.exceptions import ErrorCode, InvalidJSONError, JsonableError
 from zerver.lib.types import Validator, ViewFuncT
+from zerver.models import Client, Realm
+
+
+@dataclass
+class ZulipRequestData:
+    """This contains the additional attributes for a request that we
+    add in zerver/lib/middleware.py. Presumably, certain fields are
+    expected to not None in `request_data_map[request]` as long as
+    the request is being accessed after being processed in the middleware.
+
+    """
+
+    client: Optional[Client] = None
+    client_name: Optional[str] = None
+    client_version: Optional[str] = None
+    _log_data: Optional[MutableMapping[str, Any]] = None
```

Comment:
We might as well drop all the leading underscores as we do this migration.

Reply:
Sure